tools/ansible/tasks/wsu/test/e2e/wsu_test.go

```diff
@@ -0,0 +1,122 @@
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/masterzen/winrm"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// Get kubeconfig, AWS credentials, and artifact dir from environment variable set by the OpenShift CI operator.
+	kubeconfig     = os.Getenv("KUBECONFIG")
+	awscredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	dir            = os.Getenv("ARTIFACT_DIR")
+	privateKeyPath = os.Getenv("KUBE_SSH_KEY_PATH")
+
+	// Path of the WSU playbook
+	playbookPath = os.Getenv("WSU_PATH")
+	// clusterAddress is the address of the OpenShift cluster e.g. "foo.fah.com".
+	// This should not include "https://api-" or a port
+	clusterAddress = os.Getenv("CLUSTER_ADDR")
+
+	// The CI-operator uses AWS region `us-east-1` which has the corresponding image ID: ami-0b8d82dea356226d3 for
+	// Microsoft Windows Server 2019 Base with Containers.
+	imageID      = "ami-0b8d82dea356226d3"
+	instanceType = "m4.large"
+	sshKey       = "openshift-dev"
+
+	// Cloud provider factory that we will use in these tests
+	cloud cloudprovider.Cloud
+	// Credentials for a spun up instance
+	createdInstanceCreds *types.Credentials
+	// Temp directory ansible created on the windows host
+	ansibleTmpDir = ""
+)
+
+// createAWSWindowsInstance creates a windows instance and populates the "cloud" and "createdInstanceCreds" global
+// variables
+func createAWSWindowsInstance(t *testing.T) {
```

Comment:
This can return the `createdInstanceCreds` and remove the need for a global variable. Moreover you are introducing a dependency on `createdInstanceCreds` being populated as a side effect of executing this function. This is not obvious and can cause issues later

Reply:
How would you suggest I get the values to `testFilesCopied()`? It seems to me that pulling the functionality out of the function would be the only way to do that. I can do that, however I thought it was nice to have the test seperated into a different function. What seems preferable to you?

Reply:
I suggest to not test what should be tested as part of WNI. 

I see your point @sebsoto. Please make this a helper function that returns an error and move the assertions (requirements) to the test function.